### PR TITLE
Fixed Unit Tests(authentication.framework)

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
Fixed the unit tests for following classes
```
ApplicationAuthenticationService.class
UserSessionDAOImpl.class
SessionExtenderRequestFactory.class
```

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/resources/dbScripts/h2.sql [hunk 1]
```diff
@@ -79,26 +79,54 @@ CREATE TABLE IF NOT EXISTS IDN_OAUTH2_ACCESS_TOKEN (
...
+            IDP_ID INTEGER DEFAULT -1 NOT NULL,
+            TOKEN_BINDING_REF VARCHAR (32) DEFAULT 'NONE',
             PRIMARY KEY (TOKEN_ID),
             FOREIGN KEY (CONSUMER_KEY_ID) REFERENCES IDN_OAUTH_CONSUMER_APPS(ID) ON DELETE CASCADE,
             CONSTRAINT CON_APP_KEY UNIQUE (CONSUMER_KEY_ID,AUTHZ_USER,TENANT_ID,USER_DOMAIN,USER_TYPE,TOKEN_SCOPE_HASH,
-                                           TOKEN_STATE,TOKEN_STATE_ID)
+                                           TOKEN_STATE,TOKEN_STATE_ID,IDP_ID,TOKEN_BINDING_REF)
 );
 
-CREATE INDEX IDX_AT_CK_AU ON IDN_OAUTH2_ACCESS_TOKEN(CONSUMER_KEY_ID, AUTHZ_USER, TOKEN_STATE, USER_TYPE);
```
Comment (Thumimku) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/resources/dbScripts/h2.sql:88]: Is there a reason why we removing and adding new tables?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/ApplicationAuthenticationServiceTest.java [hunk 2]
```diff
@@ -135,18 +122,10 @@ public void testGetFederatedAuthenticators() throws Exception {
     @Test
     public void testGetRequestPathAuthenticators() throws Exception {
 
-        mockFrameworkServiceComponent();
```
Comment (Thumimku) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/ApplicationAuthenticationServiceTest.java:138]: whats the purpose of removing the mock framework component?

Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/ApplicationAuthenticationServiceTest.java:138]: For static mocking, the PowerMockTestCase/PowerMockIdentityBaseTest have to extends but it has limitation of using the initMocks() method. Therefore I remove static mocking which gives additional advantages of minimizing usage of mocking.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/resources/dbScripts/h2.sql (hunk 1 above)
Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/resources/dbScripts/h2.sql:88]: The old database did not have some required tables related to the Federated authentication. So I used an existing database which contained all required tables.

**Review by Thumimku — APPROVED** [review]

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/resources/dbScripts/h2.sql [hunk 3]
```diff
@@ -804,20 +925,291 @@ CREATE TABLE IF NOT EXISTS IDN_CERTIFICATE (
...
+    POLLING_INTERVAL INTEGER,
+    EXPIRES_IN  INTEGER,
+    AUTHENTICATED_USER_NAME VARCHAR(255),
+    USER_STORE_DOMAIN VARCHAR(100),
+    TENANT_ID INTEGER,
+    AUTH_REQ_STATUS VARCHAR (100) DEFAULT 'REQUESTED',
+    IDP_ID INTEGER,
+    UNIQUE(AUTH_REQ_ID),
+    PRIMARY KEY (AUTH_CODE_KEY),
+    FOREIGN KEY (CONSUMER_KEY) REFERENCES IDN_OAUTH_CONSUMER_APPS(CONSUMER_KEY) ON DELETE CASCADE
```
Comment (sumedhe) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/resources/dbScripts/h2.sql:962]: Shall we use similar indentations?

Comment (sumedhe) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/resources/dbScripts/h2.sql:962]: Check in other places too...

Comment (Thisara-Welmilla) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/resources/dbScripts/h2.sql:962]: Addressed with commit 930e08d24d16e5479553cf6decc257e2bc0f4bb3 in https://github.com/wso2/carbon-identity-framework/pull/3774 PR

**Pull request merged**